### PR TITLE
Add webconfiguration settings management (module + state).

### PR DESCRIPTION
**Pull request by tlemarchand**
### What does this PR do?
Add webconfiguration settings management, inside win_iis module ( get and set webconfigurations ) and win_iis state ( define webconfigurations ).
This is necessary to manage all possible IIS configurations.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/48792

### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

File: salt/modules/win_iis.py [hunk 1]
```diff
@@ -160,6 +163,37 @@ def _srvmgr(cmd, return_json=False):
...
+    collection = get_webconfiguration_settings(pspath, [{'name': name, 'filter': colfilter}])[0]['value']
+    for idx, collect_dict in enumerate(collection):
+        if all(item in collect_dict.items() for item in match.items()):
+            return idx
+    return -1
+
+
+def _prepare_settings(pspath, settings):
+    '''
+    Prepare settings before execution wit get or set functions.
```
Comment (twangboy) [inline salt/modules/win_iis.py]: with

**Review by twangboy — APPROVED** [review]

**Review by twangboy — CHANGES_REQUESTED** [review]

File: salt/modules/win_iis.py [hunk 2]
```diff
@@ -1985,3 +2019,171 @@ def set_webapp_settings(name, site, settings):
...
+
+    Returns:
+        dict: A list of dictionaries containing setting name, filter and value.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' win_iis.get_webconfiguration_settings name='IIS:\' settings="[{'name': 'enabled', 'filter': 'system.webServer/security/authentication/anonymousAuthentication'}]"
+    '''
```
[the remark this answers is not in the document]
Comment (tlemarchand) [reply salt/modules/win_iis.py]: r added line 2025?

**Review by twangboy — APPROVED** [review]

File: salt/modules/win_iis.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (twangboy) [reply salt/modules/win_iis.py]: Oh, yeah, that's right

File: salt/modules/win_iis.py [hunk 3]
```diff
@@ -160,6 +163,37 @@ def _srvmgr(cmd, return_json=False):
...
+
+
+def _prepare_settings(pspath, settings):
+    '''
+    Prepare settings before execution wit get or set functions.
+    Removes settings with a match parameter when index is not found.
+    '''
+    prepared_settings = []
+    for setting in settings:
+        match = re.search('Collection\[(\{.*\})\]', setting['name'])
```
Comment (twangboy) [inline salt/modules/win_iis.py]: This line is failing lint as well... Might need to add an `r` here. Not sure how that would affect the regex though. You'll need to test.

Comment (twangboy) [reply salt/modules/win_iis.py]: If it must be formatted like this, then you'll need to skip the lint test for this line by adding this to the end:
```
# pylint: disable=W1401
```
There should be two spaces between the end of the code and the `#`

**Review by twangboy — CHANGES_REQUESTED** [review]

File: salt/modules/win_iis.py (hunk 3 above)
Comment (tlemarchand) [reply salt/modules/win_iis.py]: The backslashes here are not interpreted by Python in any way ( neither \[ nor \{  means something ). So it is safe to add the raw prefix. I tested it, it works.

**Review by twangboy — APPROVED** [review]

**Comment by twangboy** [comment]
@rallytime Could we get this in to Fluorine? It fixes a customer issue.

**Comment by twangboy** [comment]
https://github.com/saltstack/salt/issues/48792

**Comment by twangboy** [comment]
@tlemarchand Great work! Thanks for the PR.

File: salt/modules/win_iis.py (hunk 1 above)
Comment (cachedout) [reply salt/modules/win_iis.py]: This was marked as resolved but doesn't seem to be. :)

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/modules/win_iis.py (hunk 1 above)
Comment (tlemarchand) [reply salt/modules/win_iis.py]: Good catch. Fixed now !

**Review by twangboy — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Pull request merged**